core/commands/refs.go

```diff
@@ -135,7 +135,7 @@ Note: list all refs recursively with -r.
 					return nil, errors.New(obj.Err)
 				}
 
-				return strings.NewReader(obj.Ref), nil
```

Comment:
basically, instead of putting the newline in the strings we were printing out (as seen in the root comment here: https://github.com/ipfs/go-ipfs/issues/1477), we print the newlines in the output marshaller. So the output looks exactly the same, and the json objects dont have weird newlines embedded in them.